src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1062,9 +1062,17 @@ private function createService(Definition $definition, array &$inlineServices, b
                 $callable[0] instanceof Reference
                 || $callable[0] instanceof Definition && !isset($inlineServices[spl_object_hash($callable[0])])
             )) {
-                $initializer = function () use ($callable, &$inlineServices) {
-                    return $this->doResolveServices($callable[0], $inlineServices);
-                };
+                if ($callable[0] instanceof Definition && $callable[0]->isShared()) {
```

Comment:
if `$callable[0]` is a reference, shouldn't we also check whether this reference targets a shared service ?